tests/cargo-kani/firecracker-block-example/src/main.rs

```diff
@@ -0,0 +1,440 @@
+    type CheckerType = NullChecker;
+    fn check_on_read_val(_mem: &GuestMemoryMmap, _read_val: &RequestHeader) {}
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum RequestType {
+    In,
+    Out,
+    Flush,
+    GetDeviceID,
+    Unsupported(u32),
+}
+
+pub const VIRTIO_BLK_T_IN: u32 = 0;
+pub const VIRTIO_BLK_T_OUT: u32 = 1;
+pub const VIRTIO_BLK_T_FLUSH: u32 = 4;
+pub const VIRTIO_BLK_T_GET_ID: u32 = 8;
+
+impl From<u32> for RequestType {
+    fn from(value: u32) -> Self {
+        match value {
+            VIRTIO_BLK_T_IN => RequestType::In,
+            VIRTIO_BLK_T_OUT => RequestType::Out,
+            VIRTIO_BLK_T_FLUSH => RequestType::Flush,
+            VIRTIO_BLK_T_GET_ID => RequestType::GetDeviceID,
+            t => RequestType::Unsupported(t),
+        }
+    }
+}
+
+// Kani change: simplify error type to be a simple enum
+#[derive(Debug)]
+pub enum Error {
+    /// Guest gave us too few descriptors in a descriptor chain.
+    DescriptorChainTooShort,
+    /// Guest gave us a descriptor that was too short to use.
+    DescriptorLengthTooSmall,
+    /// Getting a block's metadata fails for any reason.
+    GetFileMetadata, /*(std::io::Error)*/
+    /// Guest gave us bad memory addresses.
+    GuestMemory, /*(GuestMemoryError)*/
+    /// The data length is invalid.
+    InvalidDataLength,
+    /// The requested operation would cause a seek beyond disk end.
+    InvalidOffset,
+    /// Guest gave us a read only descriptor that protocol says to write to.
+    UnexpectedReadOnlyDescriptor,
+    /// Guest gave us a write only descriptor that protocol says to read from.
+    UnexpectedWriteOnlyDescriptor,
+    // Error coming from the IO engine.
+    FileEngine, /*(io::Error)*/
+    // Error manipulating the backing file.
+    BackingFile, /*(std::io::Error)*/
+    // Error opening eventfd.
+    EventFd, /*(std::io::Error)*/
+    // Error creating an irqfd.
+    IrqTrigger, /*(std::io::Error)*/
+    // Error coming from the rate limiter.
+    RateLimiter, /*(std::io::Error)*/
+    // Persistence error.
+    Persist, /*(crate::virtio::persist::Error)*/
+}
+
+unsafe impl kani::Invariant for Error {
+    fn is_valid(&self) -> bool {
+        matches!(
+            *self,
+            Error::DescriptorChainTooShort
+                | Error::DescriptorLengthTooSmall
+                | Error::GetFileMetadata
+                | Error::GuestMemory
+                | Error::InvalidDataLength
+                | Error::InvalidOffset
+                | Error::UnexpectedReadOnlyDescriptor
+                | Error::UnexpectedWriteOnlyDescriptor
+                | Error::FileEngine
+                | Error::BackingFile
+                | Error::EventFd
+                | Error::IrqTrigger
+                | Error::RateLimiter
+        ) || matches!(*self, Error::Persist)
+    }
+}
+
+pub const SECTOR_SHIFT: u8 = 9;
+pub const SECTOR_SIZE: u64 = (0x01_u64) << SECTOR_SHIFT;
+pub const VIRTIO_BLK_ID_BYTES: u32 = 20;
```

Comment:
I thought I put them there to mirror where they are in the original code, but I think you're right. These can move to the supporting constants file.

Reply:
I decided to leave them as is. Since this is a standalone example these constants are closer to where they are actually found in the Firecracker codebase. The virtio consts do need to be in a separate file because we use them in the finite-state-machine checker.